Jenkinsfile.windows

```diff
@@ -0,0 +1,281 @@
+@Library('pipeline-shared-libraries@feature/windows-pbp') _
+def utils
+
+pipeline {
+
+  agent { label "docker" }
+
+  options {
+    timestamps()
+    disableConcurrentBuilds()
+    buildDiscarder(
+      logRotator(
+        artifactDaysToKeepStr: '',
+        artifactNumToKeepStr: '',
+        daysToKeepStr: '',
+        numToKeepStr: '100'
+      )
+    )
+  }
+
+  parameters {
+    string(name: 'SLACK_CHANNEL', defaultValue: '#ide-builds', description: 'Slack channel to publish build message.')
+    string(name: 'RSTUDIO_VERSION_PATCH',  defaultValue: '999', description: 'RStudio Patch Version')
+    booleanParam(name: 'RSTUDIO_SKIP_QT', defaultValue: false, description: 'Skips installing and bulding for QT')
+    booleanParam(name: 'DAILY', defaultValue: false, description: 'Runs daily build if true')
+    booleanParam(name: 'PUBLISH', defaultValue: true, description: 'Runs publish stage if true')
+    gitParameter defaultValue: "${env.GIT_BRANCH}", name: 'COMMIT_HASH', type: 'PT_REVISION'
+  }
+
+  environment {
+    PACKAGE_OS = 'Windows'
+    AWS_ACCOUNT_ID = '749683154838'
+  }
+
+  stages {
+
+    stage('Load Utils') {
+      steps {
+        script {
+          sh 'printenv'
+          sh "echo 'Loading utils from ${env.WORKSPACE}/utils.groovy'"
+          utils = load "${env.WORKSPACE}/utils.groovy"
+        }
+      }
+    }
+
+    stage('Versioning') {
```

Comment:
Since versioning is based off which commit we have checked out, it's important that we have the custom checkout stage for `COMMIT_HASH` before we run the versioning step.

Reply:
That' extra painful. We'll have to use the windows agent for that stage, and then swap back to a docker agent, and then swap back again to a windows agent. We may just have to bite the bullet and write a `cmd` script for this.

Reply:
Is there a reason we can't just have a second checkout stage on the docker agent before versioning?

Reply:
Nop. We can do that. It's only a minor time cost.